client/app/authors/[authorId]/update/page.js

```diff
@@ -0,0 +1,87 @@
+"use client";
+import Link from "next/link";
+import { AuthorForm } from "../../../../components/AuthorForm";
+import { useMutation, useQuery } from "@apollo/client";
+import { use } from "react";
+import { useRouter } from "next/navigation";
+import { GenericLoader } from "../../../../components/GenericLoader";
+import { GenericError } from "../../../../components/GenericError";
+import toast, { Toaster } from "react-hot-toast";
+import dayjs from "dayjs";
+import {
+  GET_AUTHOR_BY_ID_QUERY,
+  UPDATE_AUTHOR_MUTATION,
+} from "../../../../graphql/authorGqlStrs";
+
+export default function UpdateAuthor({ params }) {
+  const unwrappedParams = use(params);
+  const { authorId } = unwrappedParams;
+  const router = useRouter();
+
+  const getAuthorMetaData = useQuery(GET_AUTHOR_BY_ID_QUERY, {
+    variables: { getAuthorById: parseInt(authorId), skip: !authorId },
+  });
+
+  const author = getAuthorMetaData?.data?.getAuthorById ?? {};
+
+  const [updateAuthor, updateAuthorMetaData] = useMutation(
+    UPDATE_AUTHOR_MUTATION
+  );
+
+  const handleSubmit = (payload) => {
+    updateAuthor({
+      variables: {
+        payload,
+        id: author.id,
+      },
+    })
+      .then(() => {
```

Comment:
We can avoid `then` & `catch`, should ideally use `async` & `await`

Reply:
used everywhere async/await instead of then/catch